From d719de0d2f3cbdb5890a147b90a51c5eaaef103e Mon Sep 17 00:00:00 2001
Message-Id: <d719de0d2f3cbdb5890a147b90a51c5eaaef103e.1650799481.git.geliang.tang@suse.com>
From: Yucong Sun <sunyucong@gmail.com>
Date: Wed, 6 Oct 2021 11:56:11 -0700
Subject: [PATCH] selftests/bpf: Fix race condition in enable_stats
Git-commit: d719de0d2f3cbdb5890a147b90a51c5eaaef103e
Patch-mainline: v5.16-rc1
References: bsn#127
Modified-by-SEL: No


In parallel execution mode, this test now need to use atomic operation
to avoid race condition.

Signed-off-by: Yucong Sun <sunyucong@gmail.com>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Link: https://lore.kernel.org/bpf/20211006185619.364369-7-fallentree@fb.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/testing/selftests/bpf/progs/test_enable_stats.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/testing/selftests/bpf/progs/test_enable_stats.c b/tools/testing/selftests/bpf/progs/test_enable_stats.c
index 01a002ade529..1705097d01d7 100644
--- a/tools/testing/selftests/bpf/progs/test_enable_stats.c
+++ b/tools/testing/selftests/bpf/progs/test_enable_stats.c
@@ -13,6 +13,6 @@ __u64 count = 0;
 SEC("raw_tracepoint/sys_enter")
 int test_enable_stats(void *ctx)
 {
-	count += 1;
+	__sync_fetch_and_add(&count, 1);
 	return 0;
 }
-- 
2.26.2

